src/lib/wp-versions.ts

```diff
@@ -97,3 +99,35 @@ export async function purgeWpConfig( wpVersion: string ) {
 
 	await fs.promises.unlink( wpConfigPath );
 }
+
+/**
+ * Verifies the checksums of a WordPress installation.
+ * If checksums don't match, it will retry the download once.
+ *
+ * @param wpVersion - The WordPress version to verify
+ */
+export async function verifyWordPressChecksums( wpVersion: string ): Promise< void > {
+	if ( ! net.isOnline() ) {
+		console.log( 'Skipping WordPress checksum verification - offline mode' );
+		return;
+	}
+
+	try {
+		console.log( `Verifying checksums for WordPress version ${ wpVersion }...` );
+		const result = await executeWPCli( getWordPressVersionPath( wpVersion ), [
+			'core',
+			'verify-checksums',
+			'--skip-plugins',
+			'--skip-themes',
+		] );
+
+		if ( result.exitCode !== 0 ) {
+			console.log( `Checksums don't match for WordPress ${ wpVersion }, retrying download...` );
+			await fs.remove( getWordPressVersionPath( wpVersion ) );
+			// Retry download one more time
+			await downloadWordPress( wpVersion, { overwrite: true } );
+		}
+	} catch ( error ) {
+		console.error( `Failed to verify WordPress checksums:`, error );
```

Comment:
If there are already issues with checksum and if downloading Wordpress is failed, probably we should not mute the error of downloading and throw it outside, so that it could be watched [here](https://github.com/Automattic/studio/blob/trunk/src/ipc-handlers.ts#L186).
WDYT?
```suggestion
		console.error( `Failed to verify WordPress checksums:`, error );
		throw error;
```

Reply:
I have tested - we indded have godd handling erors here and if we add `throw error;` then we will get this result in UI: <img width="1042" alt="Screenshot 2025-05-14 at 14 18 16" src="https://github.com/user-attachments/assets/0fdb2762-0930-4a8c-be44-c9113d5fdac9" />

So I would throw error outside to avoid muting it.

Reply:
Nice, I will make adjustments based on your suggestion 👍 It would definitely to avoid muting it. 
Yeah, I discovered a couple of strange things with downloading WordPress so I am slowly uncovering those 😅

Reply:
@nightnei I think actually your initial suggestion to throw the error inside try/catch was good. I found that we were swallowing this error later in the try/catch in `site-server.ts`. I fixed it there and now the error is properly propagating with your initial suggestion. You can test it by applying this diff: 


```diff --git a/src/lib/wp-versions.ts b/src/lib/wp-versions.ts
index ffa2033d..b19332e5 100644
--- a/src/lib/wp-versions.ts
+++ b/src/lib/wp-versions.ts
@@ -114,6 +114,7 @@ export async function verifyWordPressChecksums( wpVersion: string ): Promise< vo
 
        try {
                console.log( `Verifying checksums for WordPress version ${ wpVersion }...` );
+               throw new Error( 'Forced error for testing checksum verification' );
                const result = await executeWPCli( getWordPressVersionPath( wpVersion ), [
                        'core',
                        'verify-checksums',
```